src/coreclr/tools/Common/TypeSystem/IL/Stubs/PInvokeLazyFixupField.cs

```diff
@@ -32,6 +32,12 @@ public MethodDesc TargetMethod
             }
         }
 
+        public MethodSignature NativeSignature
+        {
+            get;
+            set;
```

Comment:
Can this be passed to the constructor? We have plenty of prior art for similar situations:

https://github.com/dotnet/runtime/blob/3eb8c7f1086b79b28a27b57a935f97be3b7fcccb/src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/DependencyAnalysis/NodeFactory.cs#L1500-L1517

Above struct contains `TypePreinit.ISerializableReference SerializableObject` but it's not part of equality/hashcode computation. It just smuggles the value through the hash table.

I don't think we have prior art for mutable type system entities on the other hand.

Reply:
I didn’t find a way to do that with `LockFreeReaderHashtable` (see comments above). I tried several but it always felt odd.

Reply:
> I didn’t find a way to do that with `LockFreeReaderHashtable` (see comments above). I tried several but it always felt odd.

Does this not work? https://github.com/dotnet/runtime/commit/60f1f177d86ab5a088ab08ca09f9a97d884d3c04

Reply:
https://github.com/dotnet/runtime/pull/99372#discussion_r1518461642

Reply:
> [#99372 (comment)](https://github.com/dotnet/runtime/pull/99372#discussion_r1518461642)

That comment was on the `record` that was making the "passenger field" part of the key because that's how records work.

What I linked uses it as a passenger, exactly how it's used in the https://github.com/dotnet/runtime/pull/99372#discussion_r1519231996 and elsewhere within the compiler. It's not part of hash code, it's not part of equality checks.

Reply:
I misunderstood the comment then. I thought that generally should not be part of the key (although the `record` part doesn't really matter here since the equality is handled in the overridden methods).

9dd48661287df65e9aed95a8802e7034efa982b1

Reply:
(Thanks. I hate this approach a bit less than the field with setter even if I would ideally prefer something similar to [ConcurrentDictionary.GetOrAdd](https://learn.microsoft.com/en-us/dotnet/api/system.collections.concurrent.concurrentdictionary-2.getoradd?view=net-8.0#system-collections-concurrent-concurrentdictionary-2-getoradd-1(-0-system-func((-0-0-1))-0)).)